test/one_hot_encoding.jl

```diff
@@ -121,6 +122,7 @@
         @test Transforms.apply(df, ohe) == expected
 
         @test Transforms.apply(df, ohe; cols=[:a]) == [[1 0 0 0 0; 0 1 0 0 0]]
```

Comment:
Wouldn't we also want this to change to not return a nested matrix?

Reply:
I thought this is what you were gonna change from your post on slack so i probably misunderstood but I would vote we don't wrap this for single cols (even if they get passed in as an array)

Reply:
My reasoning is there should still be the option to return it wrapped, and the `[:a]` vs. `:a` syntax is intuitive for that (thought come to think of it, that should be documented). Returning it wrapped keeps it consistent, e.g. if the user has some generalised code that doesn't know how many columns will be given.

Reply:
>Returning it wrapped keeps it consistent

I guess that makes sense. One thing in your favour is how extracting columns from a dataframe works.
`:a` vs `[:a]` does not return the same thing.

```julia
julia> df = DataFrame(:a=>[1, 2, 3, 4, 5], :b=>[5, 4, 3, 2, 1], :c=>[0, 1, 0, 1, 0]);

julia> df[:, :a]
5-element Array{Int64,1}:
 1
 2
 3
 4
 5

julia> df[:, [:a]]
5×1 DataFrame
│ Row │ a     │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 1     │
│ 2   │ 2     │
│ 3   │ 3     │
│ 4   │ 4     │
│ 5   │ 5     │
```

I wonder are there other examples we could lean on to inform this decision?

Reply:
Another example (note my confirmation bias):

```
julia> x = [4, 5, 6];

julia> x[1]
4

julia> x[[1]]
1-element Array{Int64,1}:
 4
```

Reply:
Example the other way:

```
julia> df = DataFrame(:a => [1, 2, 3], :b => [4, 5, 6]);

julia> combine(df, :a => sum)
1×1 DataFrame
│ Row │ a_sum │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 6     │

julia> combine(df, [:a] => sum)
1×1 DataFrame
│ Row │ a_sum │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 6     │
```

Reply:
To keep it simple and consitent I now think we should now do the :a => [...] and [:a] => [[...]] (as this PR already does)

I would argue the last example that Ben posted is actually wrapping both of them (a third option that we for sure don't want I imagine).